Block form when loading pair count

aorin opened the pull request:
Fixes an issue with water bird form where user can save a form before pair count is loaded from api.

olzraiti commented on src/components/fields/UnitCountShorthandField.js:
```diff
@@ -56,9 +57,6 @@ export default class UnitCountShorthandField extends React.Component {
 		let formData = this.props.formData;
 		const {shorthandField, pairCountField} = getUiOptions(this.props.uiSchema);
 
-		let timestamp = Date.now();
-		this.promiseTimestamp = timestamp;
```
Can you elaborate why these were removed? Now if the response for the last sent request arrives sooner than some previous one, the previous response will overwrite the last request. Some debounce mechanic would be optimal here, but that's outside of this PR's scope.

olzraiti reviewed (changes requested):
If the purpose of this PR is only preventing submitting the form, but not preventing the user interacting with the form, using a submit hook would be better (they don't show a blocker, but when user submits, the submit checks that all the submit hooks have completed before submitting). If the purpose is indeed preventing any user interaction, then this is ok (apart from the other comment I left)

aorin commented:
Didn't know about submit hooks, thanks for pointing that out. Reverted the first commit and changed it to use submit hooks. Seems to work well, does it look ok?

olzraiti reviewed (approved):
Looks good!
